packages/shared-ui/src/flowHandler/processHandler.ts

```diff
@@ -226,7 +235,7 @@ export class ProcessHandler {
     return;
   }
 
-  #updatePrimaryBlock = (newPrimaryBlock: Block<unknown>) => {
+  #updatePrimaryBlock = (newPrimaryBlock: Block<unknown>, shouldUpdateUrl = true, forcePush = false) => {
```

Comment:
Why is it called forcePush, isn't this just a normal push to history?
Also I think we should now explain a bit how we set those two values in certain situations:
- when are we setting shouldUpdateUrl to false?
- when are we using replace over push?